lib/dns.js

```diff
@@ -56,15 +56,12 @@ function errnoException(err, syscall, hostname) {
 //   callback.immediately = true;
 // }
 function makeAsync(callback) {
-  return function asyncCallback() {
+  return function asyncCallback(...args) {
     if (asyncCallback.immediately) {
       // The API already returned, we can invoke the callback immediately.
-      callback.apply(null, arguments);
+      callback.apply(null, args);
```

Comment:
Spread arguments still appear to be quite a bit slower according to benchmarks. Even the `...args` has mixed results that I haven't quite been able to track down the cause of.